tests/draft2020-12/unevaluatedItems.json

```diff
@@ -485,5 +485,183 @@
+                }
+            ],
+            "unevaluatedItems": false
+        },
+        "tests": [
+            {
+                "description": "single integer is evaluated by prefixItems",
+                "data": [ 1 ],
+                "valid": true
+            },
+            {
+                "description": "single string is evaluated by contains",
+                "data": [ "foo" ],
+                "valid": true
+            },
+            {
+                "description": "second item is evaluated by contains",
+                "data": [ 1, "foo" ],
+                "valid": true
+            },
+            {
+                "description": "second item is not evaluated, contains fails",
+                "data": [ 1, 2 ],
+                "valid": false
+            },
+            {
+                "description": "second item is not evaluated, contains passes",
+                "data": [ 1, 2, "foo" ],
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "unevaluatedItems depends on multiple nested contains",
+        "schema": {
+            "allOf": [
+                {
+                    "prefixItems": [true]
+                },
+                {
+                    "contains": {"multipleOf": 2}
+                }
+            ],
+            "anyOf": [
+                {
+                    "prefixItems": [true]
+                },
+                {
+                    "contains": {"multipleOf": 3}
+                }
+            ],
+            "unevaluatedItems": {
+                "multipleOf": 5
+            }
+        },
+        "tests": [
+            {
+                "description": "second item is evaluated by contains #1",
+                "data": [ 1, 2 ],
+                "valid": true
+            },
+            {
+                "description": "third item is evaluated by contains #2",
+                "data": [ 1, 2, 3 ],
+                "valid": true
+            },
+            {
+                "description": "items 3, 4 are evaluated by contains #2, #1",
+                "data": [ 1, 2, 3, 4 ],
+                "valid": true
+            },
+            {
+                "description": "item 5 not evaluated, passes unevaluatedItems",
+                "data": [ 1, 2, 3, 4, 5, 6 ],
+                "valid": true
+            },
+            {
+                "description": "item 5 not evaluated, fails unevaluatedItems",
+                "data": [ 1, 2, 3, 4, 7, 8 ],
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "unevaluatedItems and contains interact to control item dependency relationship",
+        "schema": {
+            "if": {
+                "contains": {"const": "a"}
+            },
+            "then": {
+                "if": {
+                    "contains": {"const": "b"}
+                },
+                "then": {
+                    "if": {
```

Comment:
This `if` doesn't do anything.  I think you just need the `const`.

Reply:
The `if` actually does play a role here. It's saying (and the logic works like this at every `if` level in the schema) that if there is a "c", that's fine and it gets evaluated and its index goes into the annotation, but if there is no "c" that's also fine - `if` doesn't assert its validity so the failure of `contains` is ignored. If we just had
```
                "then": {
                        "contains": {"const": "c"}
                }
```
in this part of the schema, it would be saying that if the instance contains any "b", then it _must_ also contain a "c". Then `[ "a", "b" ]` would fail when it should actually be valid.

Reply:
Okay, I can see that, but it doesn't read that way.

In a prod environment, I'd avoid this kind of thing and use a `properties` or something.

```
"then": {
  "properties": {
    "c": true
  }
}
```